### DISCUSSION TITLE
Bypass builtin Cpp2IL for IL2CPP games and use DLLs pre-decompiled with Il2CppDumper or other tools instead

### DISCUSSION THREAD

**Discussion by nphuracm** [AssetRipper/AssetRipper#1324] [Q&A]
I am currently working on a macOS IL2CPP game. For your information, said game is distributed as a 'Universal' binary. AssetRipper's builtin Cpp2IL logic doesn't seem to be handling this well though, as it either (on a Mac) segfaults AssetRipper altogether or (on Windows/Linux) refuses to deal with the MachO `GameAssembly.dylib` before falling AssetRipper back to "Unknown" scripting backend type.
However, there *are* tools around the corner that can indeed handle my situation pretty well, say for example Il2CppDumper. With that I am able to create DLLs that may aid in extracting actual scripts from the assets.
My question is, while this might not be exactly advisable, is it possible to use externally dumped IL2CPP DLLs for AssetRipper to extract behaviours and scripts? If so, how can this be done?

**Comment by ds5678** [comment]
Yes, just make your game look like a mono game
